service/internal/access/v2/evaluate.go

```diff
@@ -0,0 +1,321 @@
+	"github.com/opentdf/platform/protocol/go/policy"
+	attrs "github.com/opentdf/platform/protocol/go/policy/attributes"
+	"github.com/opentdf/platform/service/internal/subjectmappingbuiltin"
+	"github.com/opentdf/platform/service/logger"
+)
+
+var (
+	ErrInvalidResource              = errors.New("access: invalid resource")
+	ErrFQNNotFound                  = errors.New("access: attribute value FQN not found in memory")
+	ErrDefinitionNotFound           = errors.New("access: definition not found for FQN")
+	ErrFailedEvaluation             = errors.New("access: failed to evaluate definition")
+	ErrMissingRequiredSpecifiedRule = errors.New("access: AttributeDefinition rule cannot be unspecified")
+	ErrUnrecognizedRule             = errors.New("access: unrecognized AttributeDefinition rule")
+)
+
+// getResourceDecision evaluates the access decision for a single resource, driving the flows
+// between entitlement checks for the different types of resources
+func getResourceDecision(
+	ctx context.Context,
+	logger *logger.Logger,
+	accessibleAttributeValues map[string]*attrs.GetAttributeValuesByFqnsResponse_AttributeAndValue,
+	entitlements subjectmappingbuiltin.AttributeValueFQNsToActions,
+	action *policy.Action,
+	resource *authz.Resource,
+) (*ResourceDecision, error) {
+	if err := validateGetResourceDecision(accessibleAttributeValues, entitlements, action, resource); err != nil {
+		return nil, err
+	}
+
+	logger.DebugContext(
+		ctx,
+		"getting decision on one resource",
+		slog.Any("resource", resource.GetResource()),
+	)
+
+	switch resource.GetResource().(type) {
+	// TODO: handle registered resources
+	case *authz.Resource_RegisteredResourceValueFqn:
+		return nil, fmt.Errorf("registered resources not supported yet: %w", ErrInvalidResource)
+	case *authz.Resource_AttributeValues_:
+		return evaluateResourceAttributeValues(ctx, logger, resource.GetAttributeValues(), resource.GetEphemeralId(), action, entitlements, accessibleAttributeValues)
+
+	default:
+		return nil, fmt.Errorf("unsupported resource type: %w", ErrInvalidResource)
+	}
+}
+
+// evaluateResourceAttributeValues evaluates a list of attribute values against the action and entitlements
+// and lowercases the FQNs to ensure case-insensitive matching
+func evaluateResourceAttributeValues(
+	ctx context.Context,
+	logger *logger.Logger,
+	resourceAttributeValues *authz.Resource_AttributeValues,
+	resourceID string,
+	action *policy.Action,
+	entitlements subjectmappingbuiltin.AttributeValueFQNsToActions,
+	accessibleAttributeValues map[string]*attrs.GetAttributeValuesByFqnsResponse_AttributeAndValue,
+) (*ResourceDecision, error) {
+	// Group value FQNs by parent definition
+	definitionFqnToValueFqns := make(map[string][]string)
+	definitionsLookup := make(map[string]*policy.Attribute)
+
+	for idx, valueFQN := range resourceAttributeValues.GetFqns() {
+		// lowercase the value FQN to ensure case-insensitive matching
+		valueFQN = strings.ToLower(valueFQN)
```

Comment:
@jakedoublev Why do we have to do ToLower here versus just making sure we use `strings.EqualFold`?

Reply:
We can use `strings.EqualFold` when doing fqn and action name comparisons, but when using them as map keys in  lookups, the case sensitivity breaks when keys for the same attribute/value are of different cases. I am pushing up a change to lower case them at the highest level in the `GetDecision` flow rather than here in `evaluate`.

Reply:
If we did not do this, clients (like KAS) would need to lower case their own resource attribute value FQNs, on the way in, and it seems better to do this work at the library/auth service level for consistency.